bgpd/bgp_io.c

```diff
@@ -163,6 +164,58 @@ static void bgp_process_writes(struct thread *thread)
 	}
 }
 
+static int read_ibuf_work(struct peer *peer)
+{
+	/* static buffer for transferring packets */
+	/* shorter alias to peer's input buffer */
+	struct ringbuf *ibw = peer->ibuf_work;
+	/* packet size as given by header */
+	uint16_t pktsize = 0;
+	struct stream *pkt;
+
+	/* Hold the I/O lock, we might not have space on the InQ */
+	frr_mutex_lock_autounlock(&peer->io_mtx);
+	/* ============================================== */
+
+	if (peer->ibuf->count >= bm->inq_limit)
```

Comment:
Is there any risk of a data race here?

i.e. reading `bm->inq_limit` from the io pthread while the main pthread writes to it (upon vtysh config change)?

Reply:
yes but I elected the race is of marginal effect. the effect would only be that in this loop we read too much/too little

didn't seem worth dealing with locks for the command

Reply:
i believe we finished discussing this one via slack already